File: Roadmap/38 - MOUREDEV PRO/python/raulG91.py
import csv
import os
import random

def read_csv(filename):
    values = []
    try:
        with open(filename,newline='') as csvfile:
            reader = csv.reader(csvfile,delimiter="|")
            for index,row in enumerate(reader):
                if index != 0 and row[2].lower()=="activo":
                    values.append(row)
    except:
        print("Error opening the file")
    return values
          

#Get path for the file 
path = os.path.join(os.path.dirname(__file__),"file.csv")
#Read csv values from the file
values = read_csv(path)

if len(values)>=3:
    #Get 3 random elements from the list
    list_winners = random.sample(values,k=3)

    #First one win a subscription
    print(f'Ganadaor de la subscripcion {list_winners[0][1]} con id: {list_winners[0][0]}')

    #Second wins a discount
    print(f'Ganador del descuento {list_winners[1][1]} con id {list_winners[1][0]}')

    #Third one wins a book
    print(f'Ganador del libro {list_winners[2][1]} con id {list_winners[2][0]}')
else:
    print("Debe haber al menos 3 usuarios")

File: Roadmap/38 - MOUREDEV PRO/javascript/raulG91.js
const { stat } = require('fs')
const path = require('path')
const fs = require('fs').promises

async function readFile(filename){

    const filePath = path.join(__dirname,filename)

    let fileData =  await fs.readFile(filePath)
    //Generate a list with all rows in the file
    dataList =  fileData.toString().split("\n")
    //Remove header
    return dataList.slice(1,dataList.length)
}


async function main(){
    try{
        fileData =  await readFile('file.csv')
        let active_users = []
        for(element of fileData){
            let[id,email,status] = element.split("|")
            status = status.trim().toLowerCase()
            if(status == "activo"){
                active_users.push({"id":id,"email":email,"status":status})
            }
        }

        if(active_users.length >= 3){
            //Get subscription winner
            index = Math.floor(Math.random() * active_users.length)
            console.log("Subscription winner " + active_users[index].email + " with id: "+active_users[index].id)
            active_users.splice(index,1) //Remove selected user
            //Get discount winner

            index = Math.floor(Math.random() *active_users.length)
            console.log("Discount winner " + active_users[index].email + " with id: "+active_users[index].id)
            active_users.splice(index,1) //Remove selected user
            // Get book winner
            index = Math.floor(Math.random() *active_users.length)
            console.log("Book winner " + active_users[index].email + " with id: "+active_users[index].id)
            active_users.splice(index,1)
        }
        else{
            console.log("There are least than 3 active users")
        }

    
    }catch (error){
        console.log("Error opening the file: " + error.message)
    }
    


}

main()